src/api/XRSession.js

```diff
@@ -211,16 +211,14 @@ export default class XRSession extends EventTarget {
       return;
     }
 
-    // 'unbounded' is unlikely to ever be supported by the polyfill, since it's
-    // pretty much impossible to do correctly without native support.
-    if (type === 'unbounded') {
-      throw new NotSupportedError(`The WebXR polyfill does not support the ${type} reference space`);
-    }
-
     if (!XRReferenceSpaceTypes.includes(type)) {
       throw new TypeError(`XRReferenceSpaceType must be one of ${XRReferenceSpaceTypes}`);
     }
 
+    if (!this[PRIVATE].device.doesSessionSupportReferenceSpace(this[PRIVATE].id, type)) {
+      throw new DOMException(`The ${type} reference space is not supported by this session.`, 'NotSupportedError');
```

Comment:
Are there any scenarios where `DOMException` doesn't exist or isn't supported? [Looks good on caniuse](https://caniuse.com/#feat=mdn-api_domexception) though, so probably fine (TIL about DOMException constructor!)

Reply:
It seems widely enough supported that I'm not too worried about it.